From 51def1628a79c618b36aafc661631c8eb277ae36 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 2 Sep 2024 14:55:50 +0200
Subject: GCC: fix passing temporary array values in time_formatting.cc

GCC complains for passing a temporary array value as a parameter in
a call. To avoid that, we explicitely declare the array to pass.

Bug: 40565911
Change-Id: I56184669431832a9c8d404b714d6e22a9e380b0f
---
 base/i18n/time_formatting.cc | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/base/i18n/time_formatting.cc b/base/i18n/time_formatting.cc
index fb541d72574df..ece098967ac27 100644
--- a/base/i18n/time_formatting.cc
+++ b/base/i18n/time_formatting.cc
@@ -363,16 +363,15 @@ bool TimeDurationCompactFormatWithSeconds(TimeDelta time,
   icu::UnicodeString formatted;
   icu::FieldPosition ignore(icu::FieldPosition::DONT_CARE);
   if (hours != 0 || width == DurationFormatWidth::DURATION_WIDTH_NUMERIC) {
-    measure_format.formatMeasures(
-        (icu::Measure[3]){hours_measure, minutes_measure, seconds_measure}, 3,
-        formatted, ignore, status);
+    icu::Measure input_measures[3]{hours_measure, minutes_measure,
+                                   seconds_measure};
+    measure_format.formatMeasures(input_measures, 3, formatted, ignore, status);
   } else if (minutes != 0) {
-    measure_format.formatMeasures(
-        (icu::Measure[2]){minutes_measure, seconds_measure}, 2, formatted,
-        ignore, status);
+    icu::Measure input_measures[2]{minutes_measure, seconds_measure};
+    measure_format.formatMeasures(input_measures, 2, formatted, ignore, status);
   } else {
-    measure_format.formatMeasures((icu::Measure[1]){seconds_measure}, 1,
-                                  formatted, ignore, status);
+    icu::Measure input_measures[1]{seconds_measure};
+    measure_format.formatMeasures(input_measures, 1, formatted, ignore, status);
   }
   *out = i18n::UnicodeStringToString16(formatted);
   return U_SUCCESS(status);
-- 
2.43.0

